src/psyclone/parse/module_manager.py

```diff
@@ -109,19 +113,24 @@ def _add_all_files_from_dir(self, directory):
         module names are based on the filename using `get_modules_in_file()`.
         By default it is assumed that `a_mod.f90` contains the module `a_mod`.
 
-        :param str directory: the directory to list all files from
-        '''
+        :param str directory: the directory containing Fortran files \
+            to analyse.
 
+        '''
         with os.scandir(directory) as all_entries:
             for entry in all_entries:
                 _, ext = os.path.splitext(entry.name)
                 if (not entry.is_file()) or \
                         ext not in [".F90", ".f90", ".X90", ".x90"]:
                     continue
                 full_path = os.path.join(directory, entry.name)
+                # Obtain the names of all modules defined in this source file.
                 all_modules = self.get_modules_in_file(full_path)
                 for module in all_modules:
-                    if module not in self._mod_2_filename:
+                    # Pre-processed file should always take precedence
+                    # over non-pre-processed files:
+                    if module not in self._mod_2_filename or \
```

Comment:
I'm not feeling too sharp today so please could you explain what happens if there is a .F90?

Reply:
I tried :)